docs/ocean/design_docs/min_level_cell.rst

```diff
@@ -0,0 +1,450 @@
+several other derivative fields are initialized based on ``maxLevelCell``.
+We will rename this function ``ocn_init_routines_compute_min_max_level()``
+and will initialized the following new variables:
+
+* ``minLevelEdgeTop``
+
+* ``minLevelEdgeBot``
+
+* ``minLevelVertexTop``
+
+* ``minLevelVertexBot``
+
+The definitions of the following variables currently in this routine will be
+updated to include ``minLevelCell``:
+
+* ``boundaryEdge``
+
+* ``edgeMask``
+
+* ``cellMask``
+
+* ``vertexMask``
+
+A very large number of vertical loops will need to be altered in MPAS-Ocean.
+We will prioritize subroutines that are involved in any of the test cases for
+the ``sub_ice_shelf_2D`` configuration in legacy COMPASS.  This will give us
+a testbed for ensuring that the new functionality behaves as expected without
+requiring us to work through routines related to parameterizations, such as
+KPP and GM, that are not used in these test cases.
+
+What follows are prioritized checklists of the modules with variables that need
+to be added and subroutines where loops or other vertical indexing needs to be
+updated.  First are those related to ``init`` mode, then those used in
+``forward`` mode.
+
+Init:
+
+* [ ] ``ocn_init_sub_ice_shelf_2D``:
+
+  * [ ] ``ocn_init_setup_sub_ice_shelf_2D()``
+
+* [ ] ``ocn_init_ssh_and_landIcePressure``:
+
+  * [ ] ``ocn_init_ssh_and_landIcePressure_vertical_grid()``
+
+  * [ ] ``ocn_init_ssh_and_landIcePressure_balance()``
+
+* [ ] ``ocn_init_vertical_grids``:
+
+  * [ ] ``ocn_compute_layerThickness_zMid_from_bottomDepth()``
+
+  * [ ] ``ocn_alter_bottomDepth_for_pbcs()``
+
+  * [ ] ``ocn_compute_Haney_number()``
+
+  * [ ] ``ocn_init_vertical_grid_with_max_rx1()``
+
+  * [ ] ``constrain_rx1_layer()``
+
+Forward:
+
+* [ ] ``ocn_init_routines``:
+
+  * [ ] ``ocn_init_routines_compute_min_max_level()``
+
+  * [ ] ``ocn_init_routines_vert_coord()``
+
+  * [ ] ``ocn_init_routines_block()``
+
+* [ ] ``ocn_time_integration_split``:
+
+  * [ ] ``ocn_time_integration_split_init()``
+
+  * [ ] ``ocn_time_integrator_split()``
+
+* [ ] ``ocn_mesh``:
+
+  * [ ] ``minLevelCell``
+
+  * [ ] ``minLevelEdgeTop``
+
+  * [ ] ``minLevelEdgeBot``
+
+  * [ ] ``minLevelVertexTop``
+
+  * [ ] ``minLevelVertexBot``
+
+  * [ ] ``ocn_meshCreate()``
+
+  * [ ] ``ocn_meshUpdateFields()``
+
+  * [ ] ``ocn_meshDestroy()``
+
+* [ ] ``ocn_diagnostics``:
+
+  * [ ] ``ocn_relativeVorticity_circulation()``
+
+  * [ ] ``ocn_diagnostic_solve_layerThicknessEdge()``
+
+  * [ ] ``ocn_diagnostic_solve_vorticity()``
+
+  * [ ] ``ocn_diagnostic_solve_richardson()``
+
+  * [ ] ``ocn_diagnostic_solve_surfaceLayer()``
+
+  * [ ] ``ocn_diagnostic_solve_vortVel()``
+
+  * [ ] ``ocn_diagnostic_solve_z_coordinates()``
+
+  * [ ] ``ocn_diagnostic_solve_pressure()``
+
+  * [ ] ``ocn_vert_transport_velocity_top()``
+
+  * [ ] ``ocn_fuperp()``
+
+  * [ ] ``ocn_compute_land_ice_flux_input_fields()``
+
+  * [ ] ``ocn_validate_state()``
+
+...
```

Comment:
This list is very much incomplete and I'll be fleshing it out in the coming days.